restify/restify.d.ts

```diff
@@ -470,12 +470,12 @@ declare module "restify" {
   }
 
   interface Client {
-    get: (path: string, callback?: (err: any, req: Request, res: Response, obj: any) => any) => any;
-    head: (path: string, callback?: (err: any, req: Request, res: Response) => any) => any;
-    post: (path: string, object: any, callback?: (err: any, req: Request, res: Response, obj: any) => any) => any;
-    put: (path: string, object: any, callback?: (err: any, req: Request, res: Response, obj: any) => any) => any;
-    patch: (path: string, object: any, callback?: (err: any, req: Request, res: Response, obj: any) => any) => any;
-    del: (path: string, callback?: (err: any, req: Request, res: Response) => any) => any;
+    get: (opts: any, callback?: (err: any, req: Request, res: Response, obj: any) => any) => any;
```

Comment:
`any` feels like a too loose option - effectively you can pass anything. From documentation it seems that path is either a string or a map that might or might not have `path` + a bunch of other key value pairs. This means that parameter can be restricted to `string | { path?: string; [name: string]: any } `.